src/exercicio-2.ts

```diff
@@ -0,0 +1,84 @@
+export {};
+
+// Criação de uma interface pra referenciar como tipo do objeto permitido no array lista:
+
+interface Scientist {
+  id: number,
+  name: string,
+  bio:string
+}
+
+let lista: Scientist[] = [
+  {"id" : 1, "name": "Ada Lovelace", "bio" : "Ada Lovelace, foi uma matemática e escritora inglesa reconhecida por ter escrito o primeiro algoritmo para ser processado por uma máquina"},
+  {"id" : 2, "name": "Alan Turing", "bio" : "Alan Turing foi um matemático, cientista da computação, lógico, criptoanalista, filósofo e biólogo teórico britânico, ele é amplamente considerado o pai da ciência da computação teórica e da inteligência artificial"},
+  {"id" : 3, "name": "Nikola Tesla", "bio" : "Nikola Tesla foi um inventor, engenheiro eletrotécnico e engenheiro mecânico sérvio, mais conhecido por suas contribuições ao projeto do moderno sistema de fornecimento de eletricidade em corrente alternada."},
+  {"id" : 4, "name": "Nicolau Copérnico", "bio": "Nicolau Copérnico foi um astrônomo e matemático polonês que desenvolveu a teoria heliocêntrica do Sistema Solar."}];
+
+
+/**
+ * Encontra um objeto Scientist na lista através do id e retorna a bio do mesmo.
+ * @param list Array de Scientists declarado acima.
+ * @param id Id usado para acessar cada objeto Scientist no Array list.
+ * @returns Uma string, que será a bio do objeto Scientist encontrado, caso contrário "Nenhum cientista encontrado".
+ */
+const getScientistBio = (list: Scientist[], id: number): string => {
+  let scientist: Scientist | undefined = list.find(s => s.id === id);
+  return scientist ? scientist.bio : "Nenhum cientista encontrado";
+}
+
+console.log(getScientistBio(lista, 3))
+
+/**
+ * Encontra um objeto Scientist na lista através do id e apaga o mesmo da lista.
+ * @param list Array de Scientists declarado acima.
+ * @param id Id usado para acessar cada objeto Scientist no Array list.
+ * @returns Uma string, que será o atributo nome do objeto Scientist encontrado, caso contrário, "Nenhum cientista encontrado".
+ */
+const getScientistName = (list: Scientist[], id: number): string => {
+  let scientist: Scientist | undefined = list.find(s => s.id === id);
```

Comment:
Quando falamos de abordagem funcional, estamos falando em dizer passo a passo, por exemplo, na abordagem imperativa, não usariamos a função find, e sim utilizariamos alguma forma de for ( for in, for )

Também é preciso nomear de forma mais clara as variaveis, para ficar mais claro do que se trata.

Reply:
Corrigido.